src/app/content/highlights/components/Highlights.tsx

```diff
@@ -88,47 +88,56 @@ interface SectionHighlightsProps {
 // tslint:disable-next-line: variable-name
 export const SectionHighlights = ({ location, highlights }: SectionHighlightsProps) => {
   const pageIdIsSameAsSectionId = highlights[location.id][location.id];
+
   return (
     <React.Fragment>
       <Styled.HighlightsChapterWrapper>
         <Styled.HighlightsChapter dangerouslySetInnerHTML={{ __html: location.title }} />
       </Styled.HighlightsChapterWrapper>
-      {Object.entries(highlights[location.id]).map(([pageId, pageHighlights]) => {
-        const page = assertDefined(
-          archiveTreeSectionIsChapter(location)
-            ? findArchiveTreeNode(location, stripIdVersion(pageId))
-            : location,
-          `Page is undefined in SectionHighlights`
-        );
-        return <Styled.HighlightWrapper key={pageId}>
-          {!pageIdIsSameAsSectionId && <Styled.HighlightSection
-            dangerouslySetInnerHTML={{ __html: page.title }}
-          />}
-          {pageHighlights.map((item) => {
-            return (
-              <Styled.HighlightOuterWrapper key={item.id}>
-                <Styled.HighlightContentWrapper color={item.color}>
-                  <Styled.HighlightContent
-                    className='summary-highlight-content'
-                    data-highlight-id={item.id}
-                    dangerouslySetInnerHTML={{ __html: item.highlightedContent }}
-                  />
-                  {item.annotation ? (
-                    <Styled.HighlightNote>
-                      <span>
-                        <FormattedMessage id='i18n:toolbar:highlights:popup:body:note:text'>
-                          {(msg: Element | string) => msg}
-                        </FormattedMessage>
-                      </span>
-                      <Styled.HighlightNoteAnnotation>
-                        {item.annotation}
-                      </Styled.HighlightNoteAnnotation>
-                    </Styled.HighlightNote>
-                  ) : null}
-                </Styled.HighlightContentWrapper>
-              </Styled.HighlightOuterWrapper>
-            );
-          })}
+      {Object.entries(highlights[location.id])
+        .sort(([pageA], [pageB]) => {
+          if (archiveTreeSectionIsChapter(location)) {
+            const orderedPages = findTreePages(location);
+            return orderedPages.findIndex(({id}) => id === pageA) - orderedPages.findIndex(({id}) => id === pageB);
+          }
+          return 0;
+        })
```

Comment:
Can you move this sorting logic to `summaryHighlights` selector `src/app/content/highlights/selectors.ts`?
Also: `const orderedPages = findTreePages(location)` should be probably before `sort` so it is not recalculated for every item.

Reply:
or actually create new selector based on `summaryHighlights` so it will be recalculated only if `summaryHighlights` has changed

Reply:
you can also replace `const locationFilters = useSelector(selectors.highlightLocationFilters);` with `const locationFilters = useSelector(selectors.highlightLocationFiltersWithContent);` so we will map only for locations with highlights